nannou_laser/src/dac_manager.rs

```diff
@@ -0,0 +1,163 @@
+//! DAC detection items shared among all supported DACs
+
+use std::io;
+use thiserror::Error;
+
+pub type Result<T> = std::result::Result<T, DetectedDacError>;
+
+/// Iterators yielding laser DACs available on the system as they are discovered.
+pub enum DetectDacs {
+    EtherDream {
+        dac_broadcasts: ether_dream::RecvDacBroadcasts,
+    },
+    Helios {
+        previous_dac: helios_dac::NativeHeliosDacParams,
+    },
+}
+
+impl DetectDacs {
```

Comment:
Looking over this again, it would probably be better if `DetectDacs` and `DetectedDac` impl blocks were refactored as traits allowing for their logic to be placed into their respective files.

Reply:
Attempting to abstract each DAC seemed to add no value and in some instances, dynamic dispatch had to be used with down-casting in order retrieve the required methods needed in the stream routines. I think it's fine to leave these objects as enums rather than refactoring them with traits.